pyresample/geometry.py

```diff
@@ -892,6 +1137,239 @@ def compute_optimal_bb_area(self, proj_dict=None):
+        raise ValueError("'x' and 'y' must be an integer equal or larger than 0.")
+    if num == 0:
+        return arr
+    # Define src and dst ties
+    dst_N = (arr.size - 1) * (num + 1) + 1
+    src_ties = np.arange(dst_N, step=num + 1)
+    dst_ties = np.arange(dst_N)
+    # Interpolate
+    da = xr.DataArray(
+        data=arr,
+        dims=("x"),
+        coords={"x": src_ties}
+    )
+    da_interp = da.interp(x=dst_ties, method="linear")
+    return da_interp.data
+
+
+def _linspace2D_between_values(arr, num_x=0, num_y=0):
+    """Dask-friendly function linearly interpolating values between each 2D array values.
+
+    This function does not perform extrapolation.
+    It expects a 2D array as input!
+
+    Parameters
+    ----------
+    arr : (np.ndarray, dask.array.Array)
+        Numpy or Dask Array to be linearly interpolated between values.
+    num_x : int, optional
+        The number of linearly spaced values to infer between array values (along x).
+        . The default is 0.
+    num_y : int, optional
+        The number of linearly spaced values to infer between array values (along y).
+        The default is 0.
+
+    Returns
+    -------
+    arr : (np.ndarray, dask.array.Array)
+        Numpy or Dask Array with in-between linearly interpolated values.
+
+    Example
+    -------
+
+    Function call: _linspace1D_between_values(arr, num=1)
+    Input:
+       np.array([[5.0, 7.0],
+                 [7.0, 9.0]])
+    Output:
+        np.array([[5.0, 6.0, 7.0],
+                  [6.0, 7.0, 8.0],
+                  [7.0, 8.0, 9.0]])
+    """
+    import xarray as xr
+
+    # Check input validity
+    if arr.ndim != 2:
+        raise ValueError("'_linspace2D_between_values' expects a 2D array.")
+    num_x = int(num_x)
+    num_y = int(num_y)
+    if num_x < 0 or num_y < 0:
+        raise ValueError("'x' and 'y' must be an integer equal or larger than 0.")
+    if num_x == 0 and num_y == 0:
+        return arr
+    # Define src and dst ties
+    shape = arr.shape
+    Nx_dst = (shape[1] - 1) * (num_x + 1) + 1
+    Ny_dst = (shape[0] - 1) * (num_y + 1) + 1
+
+    src_ties_x = np.arange(Nx_dst, step=num_x + 1)
+    src_ties_y = np.arange(Ny_dst, step=num_y + 1)
+    dst_ties_x = np.arange(Nx_dst)
+    dst_ties_y = np.arange(Ny_dst)
+    # Interpolate
+    da = xr.DataArray(
+        data=arr,
+        dims=("y", "x"),
+        coords={"y": src_ties_y, "x": src_ties_x}
+    )
+    da_interp = da.interp({"y": dst_ties_y, "x": dst_ties_x}, method="linear")
+    return da_interp.data
+
+
+def _get_extended_lonlats(lon_start, lat_start, lon_end, lat_end, npts, transpose=True):
+    """Utils employed by SwathDefinition.extend.
+
+    It extrapolate npts following the forward azimuth with an interdistance
+    equal to the distance between the starting point and the end point.
+    """
+    import pyproj
+    geod = pyproj.Geod(ellps='sphere')  # TODO: sphere or WGS84?
+    #  geod = pyproj.Geod(ellps='WGS84') # sphere
```

Comment:
Why not have this as a kwarg to let the user choose?

Reply:
Done